Reuse ideas from the OpenShift Ansible guidelines, closes #2

ericzolf opened the pull request:
- add examples around line length
- speak mostly against JSON unless more readable
- add note about not defining dangerous defaults in roles
- use tasks or roles in playbooks
- naming object_action and avoid numbering
- speak against using package

richm commented on coding_style/README.adoc:
```diff
@@ -72,17 +115,17 @@ Multiple conditions that all need to be true (a logical `and`) can also be speci
-      msg: This is how it's done.
+- name: Print a message
+ansible.builtin.debug:
+msg: This is how it's done.
```
looks like the indentation was removed here?

richm commented on coding_style/README.adoc:
```diff
@@ -72,17 +115,17 @@ Multiple conditions that all need to be true (a logical `and`) can also be speci
-  - name: Print a message
-    ansible.builtin.debug: msg="This is the exact opposite of how it's done."
+- name: Print a message
+ansible.builtin.debug: msg="This is the exact opposite of how it's done."
```
looks like the indentation was removed here?

mophahr commented on coding_style/README.adoc:
```diff
@@ -96,56 +139,58 @@ Rationale:: https://yaml.org/type/bool.html[YAML 1.1] allows all variants wherea
 * Avoid comments in playbooks when possible.
-  Instead, ensure that the task `name` value is descriptive enough to tell what a task does.
-  Variables are commented in the `defaults` and `vars` directories and, therefore, do not need explanation in the playbooks themselves.
+Instead, ensure that the task `name` value is descriptive enough to tell what a task does.
```
I'm not sure if this is intentional (The indentation is off in the whole file), but I actually prefer to have lines that belong to a bullet point indeted two spaces.

mophahr commented on playbooks/README.adoc:
```diff
@@ -30,6 +30,17 @@ Examples::
 TIP: we'll explain later why there might be a case for using `include_role`/`import_role` tasks instead of the role section.
 ====
 
+== Use either tasks or roles in playbooks, not both
```
```suggestion
== Use either the tasks or roles section in playbooks, not both
```

mophahr commented on roles/README.adoc:
```diff
@@ -375,15 +375,12 @@ placed into the `vars/main.yml` file and named appropriately
-additional packages, then the required packages should be placed in `vars/main.yml` with a name such as
-`foo_packages`, and the extra packages should be passed in a variable named `foo_extra_packages`,
-which should default to an empty array in `defaults/main.yml` and be documented as such.
+Using the default filter is fine for optional keys on a dictionary, but the variable itself should be defined in `defaults/main.yml` so that it can have documentation written about it there and so that all arguments can easily be located and identified.
```
As mentioned in my other comment, I'd prefer the lines following a bullet point line to be indented ,

i.e.

```
* Bullet point 1, sentence one.
  Bullet point 1, sentence two.
* Bullet point 2.
```
instead of
```
* Bullet point 1, sentence one.
Bullet point 1, sentence two.
* Bullet point 2.
```

because I think that it is more readable.

mophahr reviewed (changes requested):
Some minor formatting questions. Other than that, this is great.

mophahr commented:
In the February 23 CoP meeting, we collectively approved the chages (up to the formatting issues already mentioned in the reviews) so this PR is ready to be merged as soon as the last open conversations are addressed/closed.

ericzolf commented:
I think I addressed all comments, and generated the docs new. The simple comment about indentation of text led to reformat the whole roles/README.adoc... Na well, it needed to be done anyway...

mophahr commented on coding_style/README.adoc:
```diff
@@ -72,20 +114,20 @@ Multiple conditions that all need to be true (a logical `and`) can also be speci
-  - name: Print a message
-    ansible.builtin.debug:
-      msg: This is how it's done.
+- name: Print a message
```
this formatting contradicts our own rules: [Indent list contents beyond the list definition](https://github.com/redhat-cop/automation-good-practices/tree/main/coding_style#yaml-and-jinja2-syntax) the same applies to l. 125 ff.

ericzolf replied:
Agreed and fixed.

mophahr commented on coding_style/README.adoc:
```diff
@@ -204,3 +249,7 @@ Rationale:: This will help with both editor highlighting as well as identifying
 ====
+
+* Using agnostic modules like `package` only makes sense if the features required are very limited.
+In many cases, if the platform is different, the package name is also different so that using `package` doesn't help a lot.
```
I'm not completely sure anymore (Gitlab diff veiw isn't great in this case) but to me it looks like this is indented diffently than the other multi-line list itrems in this file

ericzolf replied:
GitLab diff view on GitHub is indeed not great :smirk: Fixed anyway.

mophahr reviewed (changes requested):
2 indentation issues left to fix
